kani-driver/src/call_cbmc.rs

```diff
@@ -303,12 +303,31 @@ impl VerificationResult {
         }
     }
 
-    pub fn render(&self, output_format: &OutputFormat, should_panic: bool) -> String {
+    pub fn render(
+        &self,
+        output_format: &OutputFormat,
+        should_panic: bool,
+        coverage_mode: bool,
+    ) -> String {
         match &self.results {
             Ok(results) => {
                 let status = self.status;
                 let failed_properties = self.failed_properties;
                 let show_checks = matches!(output_format, OutputFormat::Regular);
+
+                // If --coverage is on, return the post-processed results
+                if coverage_mode {
+                    let mut result = formatter_coverage(
+                        results,
+                        status,
+                        should_panic,
+                        failed_properties,
+                        show_checks,
+                    );
+                    writeln!(result, "Verification Time: {}s", self.runtime.as_secs_f32()).unwrap();
+                    return result;
+                }
+
                 let mut result =
                     format_result(results, status, should_panic, failed_properties, show_checks);
```

Comment:
```suggestion
                let result = if coverage_mode {
                    formatter_coverage(
                        results,
                        status,
                        should_panic,
                        failed_properties,
                        show_checks,
                    )
                } else {
                    format_result(results, status, should_panic, failed_properties, show_checks)
                };
```

Ps.: Maybe apply the patch locally and ensure it compiles. :)